index.html

```diff
@@ -35,58 +41,150 @@
+        </aside>
+
+        The {{TCPSocket/constructor()}} steps are:
+
+        <ol>
+          <li>If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+        not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets=]", throw a
+        "{{NotAllowedError}}" {{DOMException}} and return `null`.
+          <li>If only one of |options|["{{TCPSocketOptions/localAddress}}"] or |options|["{{TCPSocketOptions/localPort}}"] is specified, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is not, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is not specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAliveDelay}}"] is less than one second, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/sendBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/receiveBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>Perform the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to open the TCP socket using the given |remoteAddress| and |remotePort| and the 
+                connection parameters (or their defaults) specified in |options|.
+              <li>
+                If this fails for any reason:
+                <ol>
+                  <li>[=Reject=] the {{TCPSocket/[[connectionPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
+                  <li>Detach {{TCPSocket/[[closedPromise]]}}.
+                </ol>
+              <li>
+                On success:
+                <ol>
+                  <li>[=initialize tcp readable stream|Initialize=] {{TCPSocket/[[readable]]}}.
+                  <li>[=initialize tcp writable stream|Initialize=] {{TCPSocket/[[writable]]}}.
+                  <li>Let |conn:TCPSocketConnection| be a new {{TCPSocketConnection}}.
+                  <li>Set |conn|["{{TCPSocketConnection/readable}}"] to [=this=].{{TCPSocket/[[readable]]}}.
+                  <li>Set |conn|["{{TCPSocketConnection/writable}}"] to [=this=].{{TCPSocket/[[writable]]}}.
+                  <li>Populate the remaining fields of |conn| using the information provided by the operating system:
+                      |conn|["{{TCPSocketConnection/remoteAddress}}"], |conn|["{{TCPSocketConnection/remotePort}}"],
+                      |conn|["{{TCPSocketConnection/localAddress}}"] and |conn|["{{TCPSocketConnection/localPort}}"].
+                  <li>Resolve [=this=].{{TCPSocket/[[connectionPromise]]}} with |conn|.
+                </ol>
+            </ol>
+          <li>Return [=this=].
```

Comment:
I don't think this is necessary. If it is please let me know where you saw this so I know what to reference in the future.

Reply:
In the blink bindings the constructor is required to return something, so I just copied the steps here. Removed :)